### ISSUE TITLE
Fix transform methods signatures

### ISSUE THREAD

**Issue by danieldietrich** [javaslang/javaslang#1100]
Our transform methods are currently not usable.

Example:

```java
// defined in Seq
default <U> U transform(Function<? super Seq<? super T>, ? extends U> f)

// defined in Option
default <U> U transform(Function<? super Option<? super T>, ? extends U> f)

...
```

**1st fix:** All transform methods have to be changed to this signature:

```java
// defined in Seq
default <U> U transform(Function<? super Seq<T>, ? extends U> f)

// defined in Option
default <U> U transform(Function<? super Option<T>, ? extends U> f)

...
```

**2nd fix:** move the transform of _Traversables_(!) to the end-points of the inheritance tree:

```java
// defined in Stream, List, Array, LinkedHashMap, HashMap, Tree, ...
default <U> U transform(Function<? super Stream<T>, ? extends U> f) { ... }
default <U> U transform(Function<? super List<T>, ? extends U> f) { ... }
...
```

![collections](https://cloud.githubusercontent.com/assets/743833/13020493/fe27d692-d1d5-11e5-8d4a-2fec2ac266ff.png)

/cc @ruslansennov I updated `U` to `? extends U` above. See #1103

**Comment by danieldietrich** [comment]
Ruslan, would be great if you could do that!

I'm finishing meanwhile the structural pattern matching.

(Currently feeling like:
![double_espresso](https://cloud.githubusercontent.com/assets/743833/13020889/22d025ba-d1d8-11e5-9e8d-45d2b137aa51.png)
)

**Comment by danieldietrich** [comment]
Fixed with #1104
